environment/src/main/java/com/sequenceiq/environment/environment/validation/cloudstorage/EnvironmentLogStorageLocationValidator.java

```diff
@@ -0,0 +1,29 @@
+package com.sequenceiq.environment.environment.validation.cloudstorage;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.util.ValidationResult;
+import com.sequenceiq.cloudbreak.util.ValidationResult.ValidationResultBuilder;
+import com.sequenceiq.environment.environment.domain.Environment;
+import com.sequenceiq.environment.environment.dto.telemetry.EnvironmentLogging;
+import com.sequenceiq.environment.environment.dto.telemetry.EnvironmentTelemetry;
+
+@Component
+public class EnvironmentLogStorageLocationValidator {
+
+    @Inject
+    private CloudStorageLocationValidator cloudStorageLocationValidator;
```

Comment:
please use ctor injection in env service

Reply:
nope

Reply:
do you have any strong objection against ctor injection in this case? if not I would ask for it also

Reply:
yes, actually I prefer explicitly expressing a framework feature (injection) rather than relying on injection by conventions

Reply:
even the authors of the framework say developers should use the ctor injection, because it's the better practice

Reply:
https://spring.io/blog/2007/07/11/setter-injection-versus-constructor-injection-and-the-use-of-required/

So what should you use?
We usually advise people to use constructor injection for all mandatory collaborators and setter injection for all other properties. Again, constructor injection ensures all mandatory properties have been satisfied, and it is simply not possible to instantiate an object in an invalid state (not having passed its collaborators). In other words, when using constructor injection you do not have to use a dedicated mechanism to ensure required properties are set (other than normal Java mechanisms).

Reply:
if a field is annotated with `@Inject` then that field is required.

Reply:
No, it means that you solely depend on the DI framework to instantiate the class and you let yourself design terrible code with monster classes. It's bad practice in every way, I don't get why people can't let this old habit go. It's not 2004 anymore. Please educate yourselves on the subject.

https://blog.marcnuri.com/field-injection-is-not-recommended/

And I repeat myself: even the authors of the framework recommend you to use constructor injection.

Reply:
I have the right to disagree but whatever. I replace the @injects.